modules/cache.py

```diff
@@ -121,3 +121,50 @@ def cached_data_for_file(subsection, title, filename, func):
         dump_cache()
 
     return entry['value']
+
+
+def prune_unused_hash():
+    import glob
+
+    from modules.paths_internal import extensions_dir
+
+    existing_cache = cache('extensions-git')
+    total_count = len(existing_cache)
+    with tqdm.tqdm(total=total_count, desc='pruning extensions') as progress:
+        for name in existing_cache:
+            if not os.path.isdir(os.path.join(extensions_dir, name)):
+                existing_cache.pop(name)
+            progress.update(1)
+
+    def file_exists(parent_dir, filename):
+        matches = glob.glob(os.path.join(parent_dir, '**', f'{filename}*'), recursive=True)
+        return len(matches) > 0
+
+    from modules.paths_internal import models_path
+    from modules.shared import cmd_opts
+
+    for db in ('hashes', 'hashes-addnet', 'safetensors-metadata'):
+        existing_cache = cache(db)
+        total_count = len(existing_cache)
+        with tqdm.tqdm(total=total_count, desc=f'pruning {db}') as progress:
+            for name in existing_cache:
+                if '/' not in name:
+                    progress.update(1)
+                    continue
+
+                category, filename = name.split('/', 1)

```

Comment:
this is wrong

- completely forgot about `hypernet`

- you are assuming that the user is using the default directories which may not be the case, that the path of each model directory is a complicated mess, some models such as Lora can come from multiple directories

they could also be special cases where extensions would add extra cache entries
and I don't think there's any way that you can account for this case

as you said
> which can reduce file size by a tiny amount, and improve load time by an even tinier amount...

personally I wouldn't bother